### DISCUSSION TITLE
add more grid land colors

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#17723] [Feature requests] [answered]
now :
![asddasads](https://user-images.githubusercontent.com/94884086/183240609-c57155de-52df-4d62-b034-9ac79a47db06.png)
 proposal : 
![blue](https://user-images.githubusercontent.com/94884086/183240775-60172cfa-4454-45b4-8210-b5d38d79f5cc.png)
![black](https://user-images.githubusercontent.com/94884086/183240827-2460027e-1a6f-4c5b-a94b-191930b2f480.png)

**Comment by karst** [comment]
You can always make a PR on https://github.com/OpenRCT2/objects adding this. If they don't want it there just make it a custom object. I doubt that this is too wanted as the devs dont want to bloat the download file.

**Reply by 733737** [reply]
it is 1kb in size, surely 1 or 2kb isn't bloat... If we added over a dozen colors, I see how that could bloat the user interface, but as it stands now there is room in the interface for 1 more color

**Reply by ocalhoun6** [reply]
Now that terrain types can be added as custom objects, why not just let these be custom objects? Anybody who wants them can easily download and install them. Or just open a park that has them already.

**Reply by 733737** [reply]
I'm aiming for official support, not having blue doesn't sit well with me, even if it is vanilla behavior. RGB is a standard for primary color systems

**Reply by ZeeMaji** [reply]
Instead of adding new grid surface objects with different color properties, what if instead terrain surfaces & edges were made recolorable ingame? Which would allow for making a single recolorable grid object to replace the currently 4 vanilla pre-defined color grid objects.
I think that would make for a better less bloaty solution that'd allow you to have the grids in _any_ color you'd want.
Would also likely be nice for custom terrain objects as they could also use it to make them more customizable.

**Reply by 733737** [reply]
I think that would be a good solution for the OpenGraphics pipeline. Right now, this cannot be implemented with the current codebase. I am not a programmer, so I couldn't personally add all 32 colors, unless each color was listed separately. In addition, I had to hand paint the preview image, as the game cannot auto generate the preview image. If someone were to program a recolorable version of this surface, they would also need to implement a new way to draw the preview image for the grid land type. Preferably also the multicolored rct1 edge types would be consolidated and recolorable. This has already been suggested in Discussions.

**Reply by ZeeMaji** [reply]
I don't really see what this would have to do with the opengraphics pipeline as all of the graphics sans the preview are already in g1.dat, as all of the grid terrain surfaces just use one set of sprites that use one of the remap colors.
That is why changing the colour property of the object changes their color, it's effectively the same as the ingame color chooser for other things albeit completely static.
For the preview image, the remap colors seem to work fine if you use it in the preview image for a terrain object (excuse the sloppy preview image, just did this as a quick test) so there doesn't seem to be any worries there.
![Screenshot_select-area_20220807100344](https://user-images.githubusercontent.com/42477864/183294638-3f88b98c-a72c-4333-a122-a8684218ec77.png)
So the only thing that'd need to be done to allow this is add recoloring functionality to the terrain tool and add the "hasPrimaryColour" flag to terrain objects to allow them to be marked as recolorable.
Along with creating a recolorable grid surface object with a proper preview to replace the original 4 preset color grid surface objects.

**Reply by 733737** [reply]
I myself don't believe I possess the skill to do all of these code modifications so I am settling for 1 color right now as a stopgap until somebody else implements recolorable land which has been discussed multiple times since last year

**Comment by 733737** [comment, marked as answer]
https://github.com/OpenRCT2/objects/pull/203

![Github Park 2022-08-06 21-28-46](https://user-images.githubusercontent.com/94884086/183272688-9bc43e6a-fc4b-423e-933b-dc490025cd69.png)
